### PR TITLE
Suggest run name based on the pipeline version used to create run

### PR DESCRIPTION
**Pull request by jingzhang36**
A couple of optimization  to save some clicks for our users when creating runs:
1. auto select default version to create run if not from any selected version's detail page
2. auto suggest a run name

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/2731)

### CODE REVIEW

File: frontend/src/pages/NewRun.test.tsx [hunk 1]
```diff
@@ -50,6 +50,10 @@ describe('NewRun', () => {
   const updateSnackbarSpy = jest.fn();
   const updateToolbarSpy = jest.fn();
 
+  const mockMath = Object.create(global.Math);
+  mockMath.random = () => 0.5;
+  global.Math = mockMath;
```
Comment (Bobgy) [inline frontend/src/pages/NewRun.test.tsx]: I think the following would be simpler
```
const randomSpy = spyOn(Math, 'random');
randomSpy.mockImplementation(() => 0.5);
```

File: frontend/src/pages/NewRun.test.tsx [hunk 2]
```diff
@@ -364,7 +382,9 @@ describe('NewRun', () => {
     expect(tree.state()).toHaveProperty('pipeline', MOCK_PIPELINE);
     expect(tree.state()).toHaveProperty('pipelineName', MOCK_PIPELINE.name);
     expect(tree.state()).toHaveProperty('pipelineVersion', MOCK_PIPELINE_VERSION);
-    expect(tree.state()).toHaveProperty('errorMessage', 'Run name is required');
+    expect((tree.state() as any).runName).toMatch(
+      /Run_of_\(original mock pipeline version name\)_.*/,
```
Comment (Bobgy) [inline frontend/src/pages/NewRun.test.tsx]: IMO, what do you think about `Run xxxx of (original mock pipeline version name)`?

There's no need for underscore and the order looks more like english?

File: frontend/src/pages/NewRun.test.tsx (hunk 1 above)
Comment (jingzhang36) [reply frontend/src/pages/NewRun.test.tsx]: Actually I tried the following and the test run hangs. That's why I used the current code snippet.

const randomSpy = jest.spyOn(Math, 'random');
randomSpy.mockImplementation(() => 0.5);

Not sure what is wrong with the above.

File: frontend/src/pages/NewRun.test.tsx (hunk 2 above)
Comment (jingzhang36) [reply frontend/src/pages/NewRun.test.tsx]: Done

File: frontend/src/pages/NewRun.test.tsx (hunk 1 above)
Comment (Bobgy) [reply frontend/src/pages/NewRun.test.tsx]: I will try to troubleshoot this together, problem with current implementation is that we don't restore the mock.

Comment (Bobgy) [reply frontend/src/pages/NewRun.test.tsx]: synced offline, seems like the following worked
```
const randomSpy = spyOn(Math, 'random');
randomSpy.mockImplementation(() => 0.5);
```

**Comment by Bobgy** [comment]
/lgtm
/approve
Thanks! This will be very convenient

**Pull request merged**